src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerTestingController.java

```diff
@@ -97,6 +101,28 @@ public CaseEventMessage getMessagesByMessageId(@PathVariable("message_id") final
         }
     }
 
+    @ApiOperation("Gets the case event message by messageId")
+    @ApiResponses({
+        @ApiResponse(
+            code = 200,
+            message = "Messages returned successfully",
+            response = CaseEventMessageEntity.class)
+    })
+    @GetMapping("/messages/query")
+    @SuppressWarnings("PMD.UseObjectForClearerAPI")
+    public EventMessageQueryResponse getMessagesByQueryParameters(
+        @RequestParam(value = "states", required = false) final String states,
+        @RequestParam(value = "case_id", required = false) final String caseId,
+        @RequestParam(value = "event_timestamp", required = false) final String eventTimestamp,
+        @RequestParam(value = "from_dlq", required = false) final Boolean fromDlq) {
```

Comment:
in this case, we cast to `Boolean` is there a specific need for this? 

Mainly saying because Boolean can be either true, false or null.
But I don't see the need for nulls so a `boolean` should suffice.
We could even add a default value to false
```suggestion
        @RequestParam(value = "from_dlq", required = false, default= false) final boolean fromDlq) {
```

Reply:
Yes, null means we don't filter on fromDlq in criteria query, so false and true would match.